### PR TITLE
[SDA-8369] Update versions to include hypershift_enabled and add validation

### PR DESCRIPTION
**Pull request by hunterkepley**
[SDA-8369](https://issues.redhat.com//browse/SDA-8369)

"Display only the versions that are hypershift supported in ocm and rosa cli"

### CODE REVIEW

**Comment by hunterkepley** [comment]
Tested with a hypershift and STS cluster

**Comment by hunterkepley** [comment]
Will un-hold after[https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/merge_requests/5401]( https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/merge_requests/5401) is merged

**Comment by gdbranco** [comment]
Seems alright, why the hold?

**Comment by oriAdler** [comment]
LGTM

File: pkg/ocm/versions.go [hunk 1]
```diff
@@ -331,3 +309,53 @@ func CheckAndParseVersion(availableUpgrades []string, version string) (string, e
...
+		allVersions := strings.Join(versionList, " ")
+		err := fmt.Errorf("A valid version number must be specified\nValid versions: %s", allVersions)
+		return version, err
+	}
+
+	if isSTS && !HasSTSSupport(version, channelGroup) {
+		err := fmt.Errorf("Version '%s' is not supported for STS clusters", version)
+		return version, err
+	}
+	if !HasSTSSupportMinor(version) {
```
Comment (oriAdler) [inline pkg/ocm/versions.go:336]: Should we check that only for **STS** clusters?

File: pkg/ocm/versions.go [hunk 2]
```diff
@@ -331,3 +309,53 @@ func CheckAndParseVersion(availableUpgrades []string, version string) (string, e
...
+			Search(filter).
+			Page(1).
+			Size(1).
+			Send()
+		if err != nil {
+			return "", handleErr(response.Error(), err)
+		}
+		valid, err := HasHostedCPSupport(response.Items().Get(0))
+		if err != nil {
+			return "", fmt.Errorf("error while parsing OCP version '%s': %v", version, err)
```
Comment (oriAdler) [inline pkg/ocm/versions.go:353]: NP
Maybe the error wrapping should be inside `HasHostedCPSupport`?

File: pkg/ocm/versions.go (hunk 1 above)
Comment (oriAdler) [reply pkg/ocm/versions.go:336]: Though I see we didn't check if it's STS previously with that `if` statement, so I'm not sure about this one.

**Review by oriAdler — COMMENTED** [review]
Left two small comments, other than that LGTM.

File: cmd/create/cluster/cmd.go [hunk 3]
```diff
@@ -1568,15 +1567,20 @@ func run(cmd *cobra.Command, _ []string) {
...
-		if subnetsProvided {
-			if isHostedCP {
-				// Hosted cluster should validate that
-				// - Public hosted clusters have at least one public subnet
-				// - Private hosted clusters have all subnets private
-				privateSubnetsCount, err = ocm.ValidateHostedClusterSubnets(awsClient, privateLink, subnetIDs)
-			} else {
-				err = ocm.ValidateSubnetsCount(multiAZ, privateLink, len(subnetIDs))
+		if subnetsProvided && !isHostedCP {
+			err = ocm.ValidateSubnetsCount(multiAZ, privateLink, len(subnetIDs))
```
Comment (gdbranco) [inline cmd/create/cluster/cmd.go:1571]: I know this was already like this, but I feel like it should be validated even if subnets are not provided. Maybe it is done like this to save a call

File: cmd/create/operatorroles/by_clusterkey.go [hunk 4]
```diff
@@ -331,6 +340,9 @@ func buildCommands(r *rosa.Runtime, env string,
 		if managedPolicies {
 			iamTags[tags.ManagedPolicies] = helper.True
 		}
+		if hostedCPPolicies {
```
Comment (gdbranco) [inline cmd/create/operatorroles/by_clusterkey.go:343]: I think this is part of Ori's PR instead

File: cmd/create/cluster/cmd.go (hunk 3 above)
Comment (oriAdler) [reply cmd/create/cluster/cmd.go:1571]: Seems like this is from Lucas PR, we need to rebase this PR against the master branch.

**Comment by davidleerh** [comment]
lgtm

**Pull request merged**